### PR TITLE
support nameSpace

### PR DESCRIPTION
**Pull request by hugosato**
The nameSpace is a config property for cache-service. This will help our test harness because the clean the keys based on nameSpace.

### CODE REVIEW

File: redisCacheModule.js [hunk 1]
```diff
@@ -277,6 +292,12 @@ function redisCacheModule(config){
     if(arguments.length < 1){
       throw new Exception('INCORRECT_ARGUMENT_EXCEPTION', '.del() requires a minimum of 1 argument.');
     }
+    if(keys === 'object') {
```
Comment (jpodwys) [inline redisCacheModule.js]: Did you mean `if(typeof keys === 'object') {`?

File: test/server/redis-cache-module.js [hunk 2]
```diff
@@ -3,7 +3,8 @@ var redisMock = require('redis-js');
 var rcModule = require('../../redisCacheModule');
 var redisCache = new rcModule({
   redisMock: redisMock,
-  backgroundRefreshInterval: 500
+  backgroundRefreshInterval: 500,
+  nameSpace: 'test'
```
Comment (jpodwys) [inline test/server/redis-cache-module.js:7]: Would you mind confirming that 'test:' shows prefixing the generated keys in a test?

File: redisCacheModule.js (hunk 1 above)
Comment (hugosato) [reply redisCacheModule.js]: ooops

File: redisCacheModule.js [hunk 3]
```diff
@@ -130,7 +143,7 @@ function redisCacheModule(config){
     }
     log(false, 'get() called:', {key: key});
     try {
-      var cacheKey = (cleanKey) ? cleanKey : key;
+      var cacheKey = (cleanKey) ? cleanKey : prefixKey(key);
```
Comment (jpodwys) [inline redisCacheModule.js]: I'm scratching my head here. I don't remember what `cleanKey` is for and I can't find a single scenario where I use it. I've looked in `cache-service-redis`, `cache-service`, `superagent-cache`, and `superagent-cache-plugin`. I'm tempted to remove it. Do you see anything that indicates why it exists?

**Comment by jpodwys** [comment]
Thanks for this, should be useful!

A note--it's possible that some users are already passing in `namespace`. As such, merging this PR will change all of their keys and may break their tests. I think we should cut this as a major release to avoid breaking anyone's deploys. Thoughts?

File: redisCacheModule.js (hunk 3 above)
Comment (hugosato) [reply redisCacheModule.js]: I've scratched my head last night trying to figure it out too...

Comment (jpodwys) [reply redisCacheModule.js]: Haha ok, let's get rid of it! You can remove it in this PR, or I can do it after merging. Doesn't matter to me.

File: redisCacheModule.js (hunk 1 above)
Comment (jpodwys) [reply redisCacheModule.js]: :)

File: test/server/redis-cache-module.js (hunk 2 above)
Comment (hugosato) [reply test/server/redis-cache-module.js:7]: So, I debated a bit around that. I think there are cases in which we want the namespace to be opaque to the cache clients, i.e.:  the namespace is associated to the backing storage only.

What do you think?

**Comment by hugosato** [comment]
Yes, I think that's a potential breaking change. It's likely a major release.

File: test/server/redis-cache-module.js (hunk 2 above)
Comment (jpodwys) [reply test/server/redis-cache-module.js:7]: I'm not sure I follow. If providing a `nameSpace` modifies the generated keys 100% of the time, it seems we should have a test that confirms it works. If I've misunderstood, please help me understand.

Comment (hugosato) [reply test/server/redis-cache-module.js:7]: Sorry, I think I misunderstood your comment. I was thinking something else related on the returned value of the prefix function... I will write the test cases.

**Comment by jpodwys** [comment]
It looks like merging this PR removed the JSON changes in your prior PR. Please submit another PR to add the JSON changes and then I'll cut a release.

**Pull request merged**